### ISSUE TITLE
Listed versions belonging to invisible or incomplete add-ons should not have (automatic) due dates

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#20198]
When the add-on is invisible (`disabled_by_user=True`) or incomplete (`status=STATUS_NULL`, generally missing metadata) its listed versions should not have a due date. They should gain one as soon as the add-on is no longer in one of these states.

**Comment by diox** [comment]
Blocks https://github.com/mozilla/addons-server/issues/19984

**Comment by wagnerand** [comment]
Note: An admin should still be able to manually set a due date for an invisible add-on. Besides that, all pending versions should be disabled automatically when the add-on is changed to that state and no new versions can be submitted.

**Comment by diox** [comment]
https://github.com/mozilla/addons-server/issues/20152 will take care of manually set due date, this issue is about the automatic ones.

> Besides that, all pending versions should be disabled automatically when the add-on is changed to that state and no new versions can be submitted.

Something to investigate here, this is already happening but only for devhub it looks like, not the API. So we should fix that if that is confirmed to be happening, and ensure listed due dates are reset in that case.

**Comment by diox** [comment]
Note that there are 2 cases to check here:
- New listed version somehow uploaded while the add-on is incomplete (likely due to lack of metadata).
- Listed version already waiting for review but somehow the add-on _becomes_ incomplete (likely due to metadata being somehow removed

I'm not sure how to get in either situation, but we should make sure they are handled properly.

**Comment by wagnerand** [comment]
See also https://github.com/mozilla/addons-server/issues/20152#issuecomment-1397177487 that covers this in part.

**Comment by diox** [comment]
```diff
diff --git a/src/olympia/versions/models.py b/src/olympia/versions/models.py
index 63c31c298c..153c9d9665 100644
--- a/src/olympia/versions/models.py
+++ b/src/olympia/versions/models.py
@@ -182,6 +182,7 @@ class VersionManager(ManagerBase):
             Q(addon__reviewerflags__auto_approval_disabled=True)
             | Q(addon__reviewerflags__auto_approval_disabled_until_next_approval=True)
             | Q(addon__promotedaddon__group_id__in=(g.id for g in PRE_REVIEW_GROUPS)),
+            addon__status__in=(amo.VALID_ADDON_STATUSES),
             channel=amo.CHANNEL_LISTED,
         )
         requires_manual_unlisted_approval_and_is_unlisted = Q(
@@ -200,7 +201,7 @@ class VersionManager(ManagerBase):
                 | requires_manual_listed_approval_and_is_listed
                 | requires_manual_unlisted_approval_and_is_unlisted
             )
-        )
+        ).using('default')
 
 
 class UnfilteredVersionManagerForRelations(VersionManager):
```

This should work for the listed version on incomplete add-on - preventing them from having a `due_date` while the add-on is incomplete. When the add-on metadata is complete, it should gain `STATUS_NOMINATED` or `STATUS_APPROVED`, triggering a reset of their due date.

**Comment by ioanarusiczki** [comment]
@diox 

I tried on stage the following scenarios with an addon flagged by scanners for manual review: 
- submitted a first version for https://reviewers.addons.allizom.org/en-US/reviewers/review/1008113. I've noticed it appeared in the Manual Queue. From dev hub checked Invisible -> the addon disappeared from queue. Submitted next versions -> same behavior.  I approved version 4.0 and submitted version 5.0 then made it invisible -> addon is no longer in queue and v5 has "Rejected or Unreviewed state"

- submitted https://reviewers.addons.allizom.org/en-US/reviewers/review/1008117 then disabled it. The addon is no longer in queue after being disabled. The next version is now awaiting review and it is present. 

- I could not submit a new version when the addon is incomplete because name or summary is missing. 
I tried for id 1008115 after deleting name/summary and changing it's state to incomplete to submit a new version using the API but I get a 400  "Add-on metadata is required to be set to create a listed version: ['name', 'summary']." -> which is the expected
